### PR TITLE
Easy renewals vanity URL

### PR DESCRIPTION
**Pull request by ScottDormand96**
https://eaflood.atlassian.net/browse/IWTF-2720?atlOrigin=eyJpIjoiMGQ1ZGVhNmZkYjE0NDdiODhlN2UxMTc3MmQzN2UyODAiLCJwIjoiaiJ9

updating to a friendly 'vanity' url for easy renewals attribution

### CODE REVIEW

File: packages/gafl-webapp-service/src/uri.js [hunk 1]
```diff
@@ -48,6 +48,7 @@ export const RENEWAL_INACTIVE = { uri: '/buy/renew/inactive', page: 'renewal-ina
 export const AUTHENTICATE = { uri: '/buy/renew/authenticate' }
 export const RENEWAL_START_DATE = { uri: '/buy/renew/renewal-start-date', page: 'renewal-start-date' }
 export const RENEWAL_START_VALIDATE = { uri: '/buy/renew/renewal-start-date/validate' }
+export const RENEWAL_LICENCE = { uri: '/renew-my-licence/{reference}', page: '/attribution?utm_campaign=renewals&urm_source=aen_invitation&reference=<licence number>' }
```
Comment (jaucourt) [inline packages/gafl-webapp-service/src/uri.js:51]: page shouldn't be defined like this - if you look, no other uri constant is set up like that. Also, unless I've missed something, you don't actually use the page property. I think you meant to, see my comment on your test, but it should either be setup in the journey definition or by using `ATTRIBUTION` above and building a query string.

File: packages/gafl-webapp-service/src/handlers/__tests__/url-handler.spec.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    ['GH330P']
+  ])('test 6 digit reference number exists and returns RENEWAL_LICENCE', async licenceKey => {
+    const query = {
+      [UTM.CAMPAIGN]: 'renewals',
+      [QUERYSTRING_LICENCE_KEY]: licenceKey
+    }
+    const responseToolkit = generateResponseToolkitMock()
+    await urlHandler(generateRequestMock(query), responseToolkit)
+    const regExMatch = new RegExp(`^${RENEWAL_LICENCE.uri}/${licenceKey}$`)
+    expect(responseToolkit.redirect).toHaveBeenCalledWith(expect.stringMatching(regExMatch))
```
Comment (jaucourt) [inline packages/gafl-webapp-service/src/handlers/__tests__/url-handler.spec.js:47]: surely it should be redirecting to the attribution page, not itself?

File: packages/gafl-webapp-service/src/handlers/url-handler.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+ */
+export default async (request, h) => {
+  await initialiseAnalyticsSessionData(request)
+  if (request.query[QUERYSTRING_LICENCE_KEY]) {
+    const refNumber = request.query[QUERYSTRING_LICENCE_KEY]
+    const sixDigit = /^[A-Za-z0-9]{6}$/.test(refNumber)
+    if (!sixDigit) {
+      return h.redirect(IDENTIFY.uri)
+    }
+    return h.redirect(`${RENEWAL_LICENCE.uri}/${refNumber}`)
```
Comment (jaucourt) [inline packages/gafl-webapp-service/src/handlers/url-handler.js:19]: the above could be simplified to:
```
if (sixDigit) {
  return h.redirect(`${RENEWAL_LICENCE.uri}/${refNumber}`)
}
```
if `sixDigit` is false, it'll fall through to the redirect below

**Review by jaucourt — CHANGES_REQUESTED** [review]

File: packages/gafl-webapp-service/src/uri.js (hunk 1 above)
Comment (ScottDormand96) [reply packages/gafl-webapp-service/src/uri.js]: ready for re review

File: packages/gafl-webapp-service/src/handlers/__tests__/url-handler.spec.js (hunk 2 above)
Comment (ScottDormand96) [reply packages/gafl-webapp-service/src/handlers/__tests__/url-handler.spec.js:47]: ready for re review

**Review by jaucourt — APPROVED** [review]

**Pull request merged**